LCLSGeneral/LCLSGeneral/POUs/Tests/FB_ECATAutoRestart_Test.TcPOU

```diff
@@ -0,0 +1,215 @@
+    fbECATAutoRestart.StateMachine(nCurrentState:=stCurrentState.deviceState,
+                                bSetError:= FALSE,
+                                bSetBusy:= FALSE,
+                                bEnable:= TRUE,
+                                nNextState => nNextState,
+                                bError=> bError);
+
+    AssertEquals_WORD(Expected:= aExpectedSeq[nIndex + 1],
+                            Actual:= nNextState,
+                            Message:= 'Did NOT Change to Expected State starting from INIT');
+END_FOR
+
+// Starting at PREOP
+FOR nIndex := 2 TO 3 DO
+    stCurrentState.deviceState := aExpectedSeq[nIndex];
+    fbECATAutoRestart.StateMachine(nCurrentState:=stCurrentState.deviceState,
+                                bSetBusy:= FALSE,
+                                bSetError:= FALSE,
+                                bEnable:= TRUE,
+                                nNextState => nNextState,
+                                bError=> bError);
+
+    AssertEquals_WORD(Expected:= aExpectedSeq[nIndex + 1],
+                            Actual:= nNextState,
+                            Message:= 'Did NOT Change to Expected State starting from PREOP');
+END_FOR
+
+// Starting in SAFEOP
+stCurrentState.deviceState := aExpectedSeq[3];
+fbECATAutoRestart.StateMachine(nCurrentState:=stCurrentState.deviceState,
+                                bSetBusy:= FALSE,
+                                bSetError:= FALSE,
+                                bEnable:= TRUE,
+                                nNextState => nNextState,
+                                bError=> bError);
+
+AssertEquals_WORD(Expected:= aExpectedSeq[4],
+                            Actual:= nNextState,
+                            Message:= 'Did NOT Change to Expected State Starting from SAFEOP');
+
+// Starting in OP
+stCurrentState.deviceState := aExpectedSeq[4];
+fbECATAutoRestart.StateMachine(nCurrentState:=stCurrentState.deviceState,
+                                bSetBusy:= FALSE,
+                                bSetError:= FALSE,
+                                bEnable:= TRUE,
+                                nNextState => nNextState,
+                                bError=> bError);
+
+AssertEquals_WORD(Expected:= aExpectedSeq[4],
+                            Actual:= nNextState,
+                            Message:= 'Did NOT Change to Expected State Starting from OP');
+
+
+
+TEST_FINISHED();]]></ST>
+      </Implementation>
+    </Method>
+    <Method Name="TestBadAddressExpectError" Id="{160b4a58-d304-419d-9ea8-d4ffaa39fa3e}">
+      <Declaration><![CDATA[
+METHOD TestBadAddressExpectError
+VAR_INPUT
+END_VAR
+VAR
+    bError				: BOOL;
+    bErrorId			: UDINT;
+
+END_VAR
+]]></Declaration>
+      <Implementation>
+        <ST><![CDATA[
+TEST('TestBadAddressExpectError');
+
+fbTON_TestTimer(IN:=bStartTest,PT:=T#5s);
+IF  NOT bStartTest THEN bStartTest := TRUE; END_IF
+
+fbECATAutoRestart(sNetId:='172.21.148.1.1',
```

Comment:
Setting a hard-coded net id in the test suite is hazardous, what if this ends up being a real device?